src/Runner.Worker/ExecutionContext.cs

```diff
@@ -1401,7 +1402,12 @@ internal TemplateTraceWriter(IExecutionContext executionContext)
 
         public void Error(string format, params Object[] args)
         {
-            _executionContext.Error(string.Format(CultureInfo.CurrentCulture, format, args));
+
+            if (logTemplateErrorsAsDebugMessages())
```

Comment:
i think the proper fix would be change `ActionsManifestManager` to properly handle exception `TemplateValidationException`

Reply:
https://github.com/actions/runner/commit/690d4f769b706f8c68632a3b3945e296dbe86a56

Reply:
When I'm reading [these lines](https://github.com/actions/runner/blob/690d4f769b706f8c68632a3b3945e296dbe86a56/src/Runner.Worker/ActionManifestManager.cs#LL133C6-L133C6):
```
catch (Exception ex) when (!(ex is TemplateValidationException))
{
                Trace.Error(ex);
                templateContext.Errors.Add(ex);
}
```
If an error is not a template validation exception add it to template context errors. Looking at just human-level language understanding I'd expect the contrary to happen, meaning add an error if the template failed validation.